cmd/kwil-admin/cmds/snapshot/create.go

```diff
@@ -233,8 +233,28 @@ func pgDump(ctx context.Context, dbName, dbUser, dbPass, dbHost, dbPort string,
 				Name:   fmt.Sprintf("validator-%d", validatorCount),
 			})
 			validatorCount++
+		} else if inFunctionBlock {
+			if strings.HasPrefix(line, "$$;") {
+				inFunctionBlock = false
+			}
+
+			// Write the line to the output file
+			n, err := multiWriter.Write([]byte(line + "\n"))
+			if err != nil {
+				return -1, nil, fmt.Errorf("failed to write to gzip writer: %w", err)
+			}
+			totalBytes += int64(n)
 		} else {
-			if line == "" || strings.TrimSpace(line) == "" { // Skip empty lines
+			if strings.HasPrefix(line, "CREATE FUNCTION") {
```

Comment:
We'd miss `CREATE OR REPLACE FUNCTION`, right?

Reply:
```
procedure create_user($name text, $age int) public {
    // we will generate a uuid from the txid
    notice('stuff $$; create function adsf() returns text AS $asd$ select 1::int8; ');
    // malice here... how?
    notice('$asd$;');
}
```

Can someone get creative and create an arbitrary *valid* function like this @brennanjl ?

Reply:
Yes. https://github.com/kwilteam/kwil-db/pull/981 fixes this. Since this edge case exists on preview right now, I think we should keep it since it is a breaking consensus change, and only put the fix on v0.9.